topics/microbiome/tutorials/dada-16S/tutorial.md

```diff
@@ -0,0 +1,1050 @@
+>    - {% icon param-file %} *"Short read data"*: `forward` output of **Unzip collection**
+>
+> 2. Rename outputs:
+>    - `dada2: learnErrors on data X, data Y, and others` to `errors for forward reads`
+>    - `dada2: learnErrors on data X, data Y, and others: error rates plot` to `error rates plot for forward reads`
+>   
+> 3. {% tool [dada2: learnErrors](toolshed.g2.bx.psu.edu/repos/iuc/dada2_learnerrors/dada2_learnErrors/1.28+galaxy0) %} with the following parameters:
+>    - {% icon param-file %} *"Short read data"*: `reverse` output of **Unzip collection**
+>
+> 4. Rename outputs:
+>    - `dada2: learnErrors on data X, data Y, and others` to `errors for reverse reads`
+>    - `dada2: learnErrors on data X, data Y, and others: error rates plot` to `error rates plot for reverse reads`
+>   
+{: .hands_on}
+
+</div>
+
+![Error rates for each possible transition as a function of the associated quality score, the final estimated error rates, the initial input rates, and the expected error rates under the nominal definition of quality scores](./images/learnErrors_forward.png "Error rates for each possible transition for the forward reads")
+
+The error rates for each possible transition (A->C, A->G, ...) are shown as a function of the associated quality score, the final estimated error rates (if they exist), the initial input rates, and the expected error rates under the nominal definition of quality scores. Points are the observed error rates for each consensus quality score. The black line shows the estimated error rates after convergence of the machine-learning algorithm. The red line shows the error rates expected under the nominal definition of the Q-score.
+
+> <question-title></question-title>
+>
+> 1. Are the estimated error rates (black line) in a good fit to the observed rates (points) for the forward reads?
+> 2. What about the reverse reads?
+>
+>    ![Error rates for each possible transition as a function of the associated quality score, the final estimated error rates, the initial input rates, and the expected error rates under the nominal definition of quality scores](./images/learnErrors_reverse.png "Error rates for each possible transition for the reverse reads")
+>
+> > <solution-title></solution-title>
+> >
+> > 1. The estimated error rates (black line) are a good fit to the observed rates (points), and the error rates drop with increased quality as expected. Everything looks reasonable and we proceed with confidence.
+> > 2. Same
+> >
+> {: .solution}
+>
+{: .question}
+ 
+
+## Infer Sample Composition 
+
+In this step, the core sample inference algorithm ({% cite Callahan_2016 %}) is applied to the filtered and trimmed sequence data.
+
+<div class="Step-by-step" markdown="1">
+
+> <hands-on-title> Sample Inference </hands-on-title>
+>
+> 1. {% tool [dada2: dada](toolshed.g2.bx.psu.edu/repos/iuc/dada2_dada/dada2_dada/1.28+galaxy0) %} with the following parameters:
+>    - *"Process samples in batches"*: `no`
```

Comment:
For advanced users we might add that choosing `yes` for `Process samples in batches` gives identical results as `no` if samples are not pooled.

- no: a single Galaxy job is started where the samples are processed sequentially
- yes: a separate Galaxy job is started for each sample (given compute resources this can be much faster)

Also the sorting of the collection is only needed for the no case.

Reply:
Added as a comment